extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -221,27 +201,37 @@ void validateBeanDeployment(
                                             .getTargetInfo()));
                 } else {
                     String channelName = instance.value().asString();
-                    Optional<AnnotationInstance> overflow = annotationStore.getAnnotations(injectionPoint.getTarget())
-                            .stream()
-                            .filter(ai -> ReactiveMessagingDotNames.LEGACY_ON_OVERFLOW
-                                    .equals(ai.name()))
-                            .filter(ai -> {
-                                if (ai.target().kind() == AnnotationTarget.Kind.METHOD_PARAMETER && injectionPoint
-                                        .isParam()) {
-                                    return ai.target().asMethodParameter().position() == injectionPoint.getPosition();
-                                }
-                                return true;
-                            })
-                            .findAny();
+                    Optional<AnnotationInstance> overflow = getOverflow(annotationStore, injectionPoint,
+                            ReactiveMessagingDotNames.LEGACY_ON_OVERFLOW);
 
                     createEmitter(emitters, injectionPoint, channelName, overflow, broadcast);
                 }
             }
         }
     }
 
+    private Optional<AnnotationInstance> getOverflow(AnnotationStore annotationStore, InjectionPointInfo injectionPoint,
```

Comment:
As this is being used to retrieve the blocking annotation as well, maybe rename to `getAnnotation` and change `onOverflowAnnotation` to `onAnnotation`?

Reply:
Fixed and rebased.